planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -1221,99 +1281,183 @@ AvoidLineArray AvoidanceModule::extractShiftLinesFromLine(ShiftLineData & shift_
   return merged_avoid_lines;
 }
 
-AvoidLineArray AvoidanceModule::applyFillGapProcess(
-  const AvoidLineArray & shift_lines, [[maybe_unused]] DebugData & debug) const
+AvoidOutlines AvoidanceModule::applyMergeProcess(
+  const AvoidOutlines & outlines, DebugData & debug) const
 {
-  AvoidLineArray ret{};
+  AvoidOutlines ret{};
 
-  if (shift_lines.empty()) {
-    return ret;
+  if (outlines.size() < 2) {
+    return outlines;
   }
 
-  const auto calc_gap_shift_line = [&](const auto & line1, const auto & line2) {
-    AvoidLine gap_filled_line{};
-    gap_filled_line.start_shift_length = line1.end_shift_length;
-    gap_filled_line.start_longitudinal = line1.end_longitudinal;
-    gap_filled_line.end_shift_length = line2.start_shift_length;
-    gap_filled_line.end_longitudinal = line2.start_longitudinal;
-    gap_filled_line.id = getOriginalShiftLineUniqueId();
+  const auto no_conflict = [](const auto & line1, const auto & line2) {
+    return line1.end_idx < line2.start_idx || line2.end_idx < line1.start_idx;
+  };
 
-    return gap_filled_line;
+  const auto same_side_shift = [](const auto & line1, const auto & line2) {
+    return line1.object_on_right == line2.object_on_right;
   };
 
-  // fill gap among shift lines.
-  for (size_t i = 0; i < shift_lines.size() - 1; i += 2) {
-    if (shift_lines.at(i).end_longitudinal > shift_lines.at(i + 1).start_longitudinal) {
+  const auto within = [](const auto & line, const size_t idx) {
+    return line.start_idx < idx && idx < line.end_idx;
+  };
+
+  ret.push_back(outlines.front());
+
+  for (size_t i = 1; i < outlines.size(); i++) {
+    auto & last_outline = ret.back();
+    auto & next_outline = outlines.at(i);
+
+    const auto & return_line = last_outline.return_line;
+    const auto & avoid_line = next_outline.avoid_line;
+
+    if (no_conflict(return_line, avoid_line)) {
+      ret.push_back(outlines.at(i));
+      continue;
+    }
+
+    const auto merged_shift_line = merge(return_line, avoid_line, getOriginalShiftLineUniqueId());
+
+    if (!isComfortable(AvoidLineArray{merged_shift_line})) {
+      ret.push_back(outlines.at(i));
```

Comment:
@satoshi-ota 
Is it okay to push back outline if `isComfortable` returns `false`?

Reply:
@rej55 Thanks for you comment!
This line checks feasibility of the merged shift line, not for the original avoid outline.
If the merged shift line is not feasible within accel/jerk constraints, skip merge process and use original avoid outline.

Reply:
@satoshi-ota Okay, I see. Thanks!